crypto/bio/bss_bio.c

```diff
@@ -267,15 +267,15 @@ static ossl_ssize_t bio_nread(BIO *bio, char **buf, size_t num_)
 
 static int bio_write(BIO *bio, const char *buf, int num_)
 {
+    if (!bio->init || buf == NULL || num_ <= 0)
```

Comment:
This cannot be moved here - we do not allow code before declarations.

Reply:
If you think it’s a reasonable change, I will change the code to satisfy the requirement. :-)

Reply:
I would say changing `num == 0` to `num_ <= 0` at the original location would be just fine.

Reply:
Totally agree. Already changed like you said.